src/factorizations/biarnoldi.jl

```diff
@@ -0,0 +1,205 @@
+# biarnoldi.jl
+mutable struct BiArnoldiFactorization{T,S} <: KrylovFactorization{T,S}
+    k::Int # current Krylov dimension
+    V::OrthonormalBasis{T} # basis of length k
+    W::OrthonormalBasis{T} # basis of length k
+    H::Vector{S} # stores the Hessenberg matrix in packed form
+    K::Vector{S} # stores the Hessenberg matrix in packed form
+    rV::T # residual
+    rW::T # residual
+end
+
+Base.length(F::BiArnoldiFactorization) = F.k
+Base.sizehint!(F::BiArnoldiFactorization, n) = begin
+    sizehint!(F.V, n)
+    sizehint!(F.W, n)
+    sizehint!(F.H, (n * n + 3 * n) >> 1)
+    sizehint!(F.K, (n * n + 3 * n) >> 1)
+    return F
+end
+Base.eltype(F::BiArnoldiFactorization) = eltype(typeof(F))
+Base.eltype(::Type{<:BiArnoldiFactorization{<:Any,S}}) where {S} = S
+
+basis(F::BiArnoldiFactorization) = (F.V, F.W)
+function rayleighquotient(F::BiArnoldiFactorization)
+    return (PackedHessenberg(F.H, F.k), PackedHessenberg(F.K, F.k))
+end
+residual(F::BiArnoldiFactorization) = (F.rV, F.rW)
+@inbounds normres(F::BiArnoldiFactorization) = (abs(F.H[end]), abs(F.K[end]))
+rayleighextension(F::BiArnoldiFactorization) = SimpleBasisVector(F.k, F.k)
+
+struct BiArnoldiIterator{F,T,O<:Orthogonalizer} <: KrylovIterator{F,T}
+    operator::F
+    v₀::T
+    w₀::T
+    orth::O
+end
+BiArnoldiIterator(f, v₀, w₀) = BiArnoldiIterator(f, v₀, w₀, KrylovDefaults.orth)
+
+Base.IteratorSize(::Type{<:BiArnoldiIterator}) = Base.SizeUnknown()
+Base.IteratorEltype(::Type{<:BiArnoldiIterator}) = Base.EltypeUnknown()
+
+function Base.iterate(iter::BiArnoldiIterator)
+    state = initialize(iter)
+    return state, state
+end
+function Base.iterate(iter::BiArnoldiIterator, state)
+    nr = normres(state)
+    if nr < eps(typeof(nr))
```

Comment:
Looking at the normres implementation on line 28, I think `nr` here is a tuple with the two norms in the final elements of H and K. Do you want both to be smaller than eps? Or just one of them? I would expect that this immediately throws an error upon the first run.